chacra/tests/async/test_recurring.py

```diff
@@ -37,22 +37,142 @@ def test_gets_rid_of_old_repos(self, session, fake, monkeypatch):
+            distro='centos',
+            distro_version='7',
+            flavor='ceph_master',
+        )
+
+        conf.purge_rotation = {'nfs-ganesha': {'ref': {'next': {'keep_minimum': 0}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 0
+
+    def test_get_rid_of_old_repos_by_flavor_configured_in_days(self, session, fake, monkeypatch):
+        Repo(
+            project=Project('nfs-ganesha'),
+            ref='next',
+            distro='centos',
+            distro_version='7',
+            flavor='ceph_master',
+        )
+
+        conf.purge_rotation = {'nfs-ganesha': {'flavor': {'ceph_master': {'days': 0}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 0
+
+    def test_get_rid_of_old_repos_by_flavor_configured_with_offset(self, session, fake, monkeypatch):
+        Repo(
+            project=Project('nfs-ganesha'),
+            ref='next',
+            distro='centos',
+            distro_version='7',
+            flavor='ceph_master',
+        )
+
+        conf.purge_rotation = {'nfs-ganesha': {'flavor': {'ceph_master': {'keep_minimum': 0}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 0
+
+    def test_does_not_get_rid_of_old_repos_by_flavor_and_ref(self, session, fake, monkeypatch):
+        conf.purge_rotation = {'ceph': {'flavor': {'default': {'keep_minimum': 1}}, 'ref': {'firefly': {'days': 70}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 1
+
+    def test_get_rid_of_old_repos_by_flavor_with_ref(self, session, fake, monkeypatch):
+        Repo(
+            project=Project('nfs-ganesha'),
+            ref='next',
+            distro='centos',
+            distro_version='7',
+            flavor='ceph_master',
+        )
+
+        conf.purge_rotation = {'nfs-ganesha': {'flavor': {'ceph_master': {'keep_minimum': 0}}, 'ref': {'next': {'days': 70}}}, '__force_dict__': True}
+        fake_datetime = fake(utcnow=lambda: self.old, now=self.now)
+        monkeypatch.setattr(datetime, 'datetime', fake_datetime)
+        session.commit()
+        recurring.purge_repos(_now=self.now)
+        assert len(Repo.query.all()) == 0
```

Comment:
If I'm understanding this right the repo here passed the test to be kept by it's ``ref`` but not it's ``flavor`` and was deleted. Is that correct? If so I'm not sure that's the behavior I'd expect. I'd think the repo should not be deleted because it should be kept around because of how it's configured for ``ref``.

Reply:
@alimaredia since you are ready for this to be re-reviewed, would you mind clarifying this point further?

Reply:
@alimaredia I think this one still needs clarification still

Reply:
This test could be renamed to `test_get_rid_of_new_repo_with_ref_bad_days_and_flavor_bad_offset`. The point of this test and the two tests after it is to test if only "bad" inputs are given to the purge rotation in various combinations and to make sure all the repos are purged. Does that help?